christfellowshipchurch/web-app-v2
# Group sub preference images

## What is this and why is it needed?
Requirement from the Groups team: when a Line Up is shown on a Hub page, the combination of Hub/Lineup should be able to have a unique image. For example, Sisterhood/Activity should have Image A and Crew/Activity should have Image B.

This is _only_ to be shown on the Community Single page (`/community/[title]`) and should also have a default image.

## How do I test this?
Right now, there are not very many groups set up, so there might not be any lineups showing up at the moment. In order to get around this, you can open up the `CommunitySingle` and comment out the line where the lineups are being filtered. I've also attached some screenshots

**This build requires a specific API branch in order to work as expected. See the related issues**

## Screenshots
![Screen Shot 2021-05-14 at 4 33 08 PM](https://user-images.githubusercontent.com/45076058/118326665-44c82a80-b4d3-11eb-8aad-7679a2e3bf04.png)
![Screen Shot 2021-05-14 at 4 33 27 PM](https://user-images.githubusercontent.com/45076058/118326672-45f95780-b4d3-11eb-96f9-70ad644c12ee.png)
![Screen Shot 2021-05-14 at 4 33 58 PM](https://user-images.githubusercontent.com/45076058/118326674-45f95780-b4d3-11eb-9c2c-213bae631738.png)

## Tradeoffs/Design Considerations
The way that this is set up allows for any Node Id to be passed in to the `coverImage` of the Preference. This may not have been 10000000% necessary, but it didn't take anymore time/effort and also allows for the possibility to filter the cover image by other node types (like a specific group, for instance)

While there are no current use cases for this, it felt like the least obtrusive way to set this up and also narrows down the "filtering" logic to _just_ the `coverImage`

## Related Issues
[CFDP-1377]
API Dependency: https://github.com/christfellowshipchurch/apollos-api/pull/249

[CFDP-1377]: https://christfellowshipchurch.atlassian.net/browse/CFDP-1377
